components/dropdown/dropdown-content-mixin.js

```diff
@@ -978,8 +1011,60 @@ export const DropdownContentMixin = superclass => class extends LocalizeCoreElem
 		return false;
 	}
 
-	_getPosition(spaceAround, targetWidth, contentWidth) {
-		const centerDelta = contentWidth - targetWidth;
+	_getPointerPosition(targetRect) {
+		const position = {};
+		if (!this._fixedPositioning) return position;
+
+		const pointer = this.__getPointer();
+		if (!pointer) return position;
+
+		const pointerRect = pointer.getBoundingClientRect();
+		const isRTL = this.getAttribute('dir') === 'rtl';
+		if (!isRTL) {
+			position.left = targetRect.left + ((targetRect.width - pointerRect.width) / 2);
+		} else {
+			position.right = window.innerWidth - targetRect.left - ((targetRect.width - pointerRect.width) / 2);
+		}
+		if (this.openedAbove) {
+			position.bottom = window.innerHeight - targetRect.top + 8;
+		} else {
+			position.top = targetRect.top + targetRect.height + this._verticalOffset - 7;
+		}
+
+		return position;
+	}
+
+	_getPosition(spaceAround, targetRect, contentRect) {
```

Comment:
As mentioned, this has been updated to return an object, rather than just the left adjustment value. The non-fixed positioning value remains the same.